<a href="https://colab.research.google.com/github/subin100/DataVisualization/blob/main/Quiz4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


In [13]:
%%bigquery --project datavisualization-338803
SELECT *
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

Limit 10

trip_id  ... duration_minutes
0  9900289692  ...               39
1  9900285987  ...               31
2  9900285989  ...               31
3  9900285991  ...               30
4  9900286140  ...               19
5  9900286143  ...               17
6  9900286171  ...                6
7  9900286214  ...                0
8  9900286540  ...               12
9    13575843  ...               47

[10 rows x 9 columns]

This gives us an idea of what is in the table to reference. I have used limit 10 which shows only 10 columns of data and also it takes less amount of time to process. I am more interested on start_station_name and duration_time from this table as we have to work mostly on these two dataset for this assignment.

In [20]:
%%bigquery --project datavisualization-338803
SELECT AVG(table.duration_minutes) as average_duration_minutes, start_station_name
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips` as table
GROUP BY start_station_name

average_duration_minutes                     start_station_name
0                   27.168259                       Zilker Park West
1                   28.115798                Toomey Rd @ South Lamar
2                   30.421842        State Capitol @ 14th & Colorado
3                   23.265843                       Waller & 6th St.
4                   28.648889                             Pease Park
..                        ...                                    ...
188                 23.067313  Republic Square @ Guadalupe & 4th St.
189                 19.937466                             3rd & West
190                 34.152547                               3rd/West
191                 16.805353                          Nueces & 26th
192                 24.521670                            26th/Nueces

[193 rows x 2 columns]

Here, we calculated average time for the trip using Avg command and grouped it in a average_duration_minutes and then we based in on starting point. For example, according to the table, it takes around 27.17 minutes to make a trip in Ziker park West. 

In [35]:
%%bigquery --project datavisualization-338803
WITH t as(
SELECT AVG(table.duration_minutes) as average_duration_minutes, start_station_name
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips` as table
GROUP BY start_station_name
)
SELECT *, 
    RANK() OVER(
      PARTITION BY start_station_name
    ORDER BY start_station_name) rn_num
FROM t
ORDER BY rn_num


average_duration_minutes                       start_station_name  rn_num
0                   34.330909               East 7th & Pleasant Valley       1
1                   28.750000               Mobile Station @ Bike Fest       1
2                   25.702632                          28th/Rio Grande       1
3                   29.479225                           8th & Congress       1
4                  102.406948                              6th/Trinity       1
..                        ...                                      ...     ...
188                 40.517433                  Congress & Cesar Chavez       1
189                 24.830726              East 6th at Robert Martinez       1
190                  6.800000                                     Shop       1
191                 28.976835  Capital Metro HQ - East 5th at Broadway       1
192                 25.552000     State Parking Garage @ Brazos & 18th       1

[193 rows x 3 columns]

From the table it seems that lots of the trips at the each starting point only once.